### ISSUE TITLE
Index out of bounds

### ISSUE THREAD

**Issue by kallestenflo** [jayway/JsonPath#79]
Given JSON:
```
{ 
  "c": {
    "d1": {
      "url": [ "url1", "url2" ]
    },
    "d2": {
      "url": [ "url3", "url4","url5" ]
    }
  }
}
```
The path:
```
$.c.*.url[2]
```
throws an exception

Index out of bounds when evaluating path $['c']['d1']['url'][2]

Expected result:

```
[
   "url5"
]
```

**Comment by Delagen** [comment]
The bug is for version 1.2 also.
JMeter-plugins made decision to revert to 0.9 branch of this library due disagree with author.
https://groups.google.com/forum/#!topic/jmeter-plugins/fDqEpL3iqi0

**Comment by kallestenflo** [comment]
Fixed for 2.1.
